src/operator/src/expr_helper.rs

```diff
@@ -287,9 +290,52 @@ pub fn validate_create_expr(create: &CreateTableExpr) -> Result<()> {
         }
     }
 
+    if auto_schema {
+        // `DateTime` type is not supported in auto schema mode.
```

Comment:
If auto schema is false, we already converted the datetime type to timestamp(3) before this function. Maybe we can also return an error?

Reply:
If the auto schema is false, it means the table was manually created. In this case, DateTime is an alias for Timestamp(6), so the actual type will be Timestamp(6). If you specify this column as `DateTime` through the gRPC protocol during writing, a type check error will occur (this check has always existed).